crates/nu-command/tests/commands/headers.rs

```diff
@@ -30,6 +30,19 @@ fn headers_adds_missing_column_name() {
     assert_eq!(actual.out, r#"["r1c1","r2c1"]"#)
 }
 
+#[test]
+fn headers_handles_missing_values() {
+    let actual = nu!(
+    cwd: "tests/fixtures/formats", pipeline(
+        r#"
+            [{x: x, y: y}, {x: x, y: y}, {x: x, z: z}]
+            | headers
+            | to json --raw"#
+    ));
+
+    assert_eq!(actual.out, r#"[{"x": "x","y": "y"},{"x": "x"}]"#)
+}
```

Comment:
```suggestion
#[test]
fn headers_handles_missing_values() {
    let actual = nu!(
    cwd: "tests/fixtures/formats", pipeline(
        "
            [{x: a, y: b}, {x: 1, y: 2}, {x: 1, z: 3}]
            | headers
            | to nuon --raw
        "
    ));

    assert_eq!(actual.out, "[{a: 1, b: 2}, {a: 1}]")
}
```

Reply:
(@amtoine you need to keep the `pipeline()` function, to wrap the pipelline over multiple lines, because `nu!` will think they are separate statements and join them with a `;`. Mildly confusing/annoying. See #9609 )

Reply:
ooooh ok i see :+1: 

then either
```rust
    let actual = nu!(pipeline("
            [{x: a, y: b}, {x: 1, y: 2}, {x: 1, z: 3}]
            | headers
            | to nuon --raw
    "));
```
or
```nushell
    let actual = nu!("[{x: a, y: b}, {x: 1, y: 2}, {x: 1, z: 3}] | headers | to nuon --raw");
```
@IanManske :wink:

Reply:
Fixed!